File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0485_Max_Consecutive_Ones.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0485_Max_Consecutive_Ones {
    public int findMaxConsecutiveOnes(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int res = 0, cnt = 0;
        for (int n : nums) {
            if (n == 1) {
                cnt++;
                res = Math.max(res, cnt);
            } else {
                cnt = 0;
            }
        }
        return res;
    }
}

File: algorithms/algorithms-cplusplus/leetcode/Question_0485_Max_Consecutive_Ones.cc
#include <vector>

using namespace std;

class Solution {
 public:
  int findMaxConsecutiveOnes(vector<int>& nums) {
    int res = 0, cnt = 0;
    for (int n : nums) {
      if (n == 1) {
        cnt++;
        res = max(res, cnt);
      } else {
        cnt = 0;
      }
    }
    return res;
  }
};
